File: 2021/12/20/main.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        md = 10**6+1
        arr.sort()
        for i in range(1, len(arr)):
            md = min(md, arr[i]-arr[i-1])
        r = []
        for i in range(1, len(arr)):
            if arr[i]-arr[i-1] == md:
                r.append([arr[i-1], arr[i]])
        return r

File: 2021/12/20/main.go
package main

import (
	"math"
	"sort"
)

func minimumAbsDifference(arr []int) [][]int {
	sort.Ints(arr)
	n := len(arr)

	minDiff := math.MaxInt64
	for i := 1; i < n; i++ {
		minDiff = min(minDiff, arr[i]-arr[i-1])
	}

	res := make([][]int, 0, n)

	for i := 1; i < n; i++ {
		if arr[i]-arr[i-1] == minDiff {
			res = append(res, []int{arr[i-1], arr[i]})
		}
	}

	return res
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}
